PresentData/PresentMonTraceConsumer.cpp

```diff
@@ -1139,6 +1188,169 @@ void PMTraceConsumer::HandleDWMEvent(EVENT_RECORD* pEventRecord)
     }
 }
 
+void PMTraceConsumer::RemovePresentFromTemporaryTrackingCollections(std::shared_ptr<PresentEvent> p)
+{
+    // Remove the present from any struct that would only host the event temporarily.
+    // Currently defined as all structures except for mPresentsByProcess, 
+    // mPresentsByProcessAndSwapChain, and mAllPresents.
+
+    // mPresentByThreadId
+    {
+        auto eventIter = mPresentByThreadId.find(p->ThreadId);
+        if (eventIter != mPresentByThreadId.end() && eventIter->second == p) {
+            mPresentByThreadId.erase(eventIter);
+        }
+        // TODO: Assert that an element has been removed if this element must exist.
+    }
+
+    // mPresentsBySubmitSequence
+    {

```

Comment:
This function is full of unnecessary scopes. I'd remove them. A scope is great when it means local variables can't leak, or to force destruction to happen earlier, but a scope around an `if` which is going to introduce its own new scope isn't useful.

Reply:
Good point, now that I look at this function again it has a lot more if conditions than it started out with, will address in the next revision.